### PR TITLE
Patch for issue command gathered when opening terminal

### PR DESCRIPTION
**Pull request by ndcroos**
https://github.com/NorfairKing/hastory/issues/3
I started looking at this bug. Instead of executing a script function before each command and with each new session, the script function is now executed after each command.

### CODE REVIEW

File: hastory-cli/src/Hastory/Cli/Utils.hs [hunk 1]
```diff
@@ -19,4 +19,4 @@ doCountsWith conv func = foldl go HM.empty
     go hm k = HM.alter a (conv k) hm
       where
         a Nothing = Just 1
-        a (Just d) = Just $ d + func k
+        a (Just d) = Just $ d + func k
```
Comment (NorfairKing) [inline hastory-cli/src/Hastory/Cli/Utils.hs]: Please don't remove newlines at the end of files.

File: hastory-cli/src/Hastory/Cli/Commands/Gather.hs [hunk 2]
```diff
@@ -31,4 +31,4 @@ storeHistory entry = do
     hFile <- histFileFor $ entryDateTime entry
     liftIO $ do
         ensureDir $ parent hFile
-        LB.appendFile (toFilePath hFile) $ JSON.encode entry <> "\n"
+        LB.appendFile (toFilePath hFile) $ JSON.encode entry <> "\n"
```
Comment (NorfairKing) [inline hastory-cli/src/Hastory/Cli/Commands/Gather.hs]: Please don't remove newlines at the end of files.

File: hastory-cli/app/Main.hs [hunk 3]
```diff
@@ -3,4 +3,5 @@ module Main where
 import Hastory.Cli
 
 main :: IO ()
-main = hastoryCli
+main = do
+  hastoryCli
```
Comment (NorfairKing) [inline hastory-cli/app/Main.hs]: This is also a strange change.

**Comment by ndcroos** [comment]
I did the requested changes.

File: README.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ This means that the only thing you need to do is to add a one-liner to the scrip
 For bash, it suffices to add the following to .bashrc.
 
 ``` shell
-PS1+='$(hastory_gather_)'
+PROMPT_COMMAND="hastory_gather_"
```
Comment (NorfairKing) [inline README.md]: You also need to update the docs for `zsh`, that now needs something like this:
```
precmd() { hastory_gather_ }
```

Comment (ndcroos) [reply README.md]: I changed the readme. With the zsh shell, the timestamps seem to be correct.

**Pull request merged**